lib/spack/spack/compilers/gcc.py

```diff
@@ -87,6 +87,24 @@ def cxx17_flag(self):
         else:
             return "-std=c++17"
 
+    @property
+    def c99_flag(self):
+        if self.version < ver('4.5'):
```

Comment:
Is this right?  Unsupported before 4.5?  AFAIK C99 was fully supported in GCC all the way back to 3.4

Reply:
For this constraint I used [this reference](https://www.gnu.org/software/gcc/c99status.html).

Reply:
I'm not sure it really matters that much since we're pretty guaranteed to have at least 4.7 from SUSE12 or 4.8 from RHEL7.